snapcraft/internal/pluginhandler/_outdated_report.py

```diff
@@ -0,0 +1,63 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from snapcraft import formatting_utils
+from snapcraft.internal import steps
+
+
+class OutdatedReport:
+    """The OutdatedReport class explains why a given step is outdated."""
```

Comment:
can you further expand in a paragraph what it means to be outdated (conditions that trigger it)? Simil for DirtyReport please.

Reply:
Done, for both outdated and dirty.